sdk/core/core-rest-pipeline/src/interfaces.ts

```diff
@@ -47,6 +47,101 @@ export interface HttpHeaders extends Iterable<[string, string]> {
   toJSON(options?: { preserveCase?: boolean }): RawHttpHeaders;
 }
 
+/**
+ * A blob that is streamable. `Blob` and `File` satisfy this interface.
+ */
+export interface StreamableBlob {
+  /**
+   * A stream (or a function that returns a stream) representing the blob's contents.
+   */
+  stream: ReadableStream | NodeJS.ReadableStream | (() => ReadableStream | NodeJS.ReadableStream);
+
+  /**
+   * The MIME type of the blob.
+   */
+  type?: string;
+
+  /**
+   * The size of the blob, if known.
+   */
+  size?: number;
+}
+
+/**
+ * Blob with content as a Uint8Array
+ */
+export interface InMemoryBlob {
+  /**
+   * The content of the blob as a Uint8Array
+   */
+  content: Uint8Array;
+
+  /**
+   * The MIME type of the blob.
+   */
+  type?: string;
+}
+
+/**
+ * A type representing a blob.
+ *
+ * `Blob` and `File` objects satisfy this, but a ReadableStream (web or Node) can be
+ * used instead by creating an object with the `stream` property set to the stream.
+ */
+export type BlobLike = StreamableBlob | InMemoryBlob;
+
+/**
+ * A type representing a file to be uploaded.
+ * The `File` class satisfies this type, allowing for instances to be passed directly. If the file to be
+ * uploaded is not represented by a `File` or `Blob`, an object containing a stream or content buffer and optional file type and file name
+ * can be used instead.
+ */
+export type FileLike = BlobLike & {
+  /**
+   * The name of the file. If no file name is present, a placeholder name of 'blob' will be used when submitting a form
+   * containing the file.
+   */
+  name?: string;
+};
+
+/**
+ * A part of the request body in a multipart request.
+ */
+export interface BodyPart {
+  /**
+   * The headers for this part of the multipart request.
+   */
+  headers: HttpHeaders;
+
+  /**
+   * The body of this multipart request.
+   */
+  body: ReadableStream | NodeJS.ReadableStream | Uint8Array | BlobLike;
```

Comment:
Should we drop `Uint8Array` from here since we already included `BlobLike`?

Reply:
Also same question about the streams.

Reply:
I prefer it as-is personally. Outside of the form data context the extra properties on `BlobLike` are not relevant (other than sometimes size) so I don't think we should force people to wrap the uint8array/stream in another object if they don't need to.

Reply:
I see, then could we make it as a function of the `BlobLike` definitions to reduce duplication and get automatic support for anything gets added in the future? e.g. `body: InMemoryBlob["content"] | StreamingBlob["stream"] | BlobLike;`

Reply:
Hmm, we could, but I feel that the benefit is marginal. I think leaving it as-is provides a cleaner API, and we won't be adding new types to BlobLike et al very often (if at all). If we were to add a new type, that would probably require code changes in the policy anyway so I consider it unlikely that we would accidentally update BlobLike but not the policy itself.

As a comparison, `RequestBodyType` already takes in `Blob | ReadableStream | NodeJS.ReadableStream | ...`. This is just following a similar pattern.

Reply:
I agree it is not a big deal either way, however, there is already stuff supported in `BlobLike` that is not supported here, resettable streams being one IIRC.

Reply:
That's only there because of the contract for Blob (`stream` is a function in that definition). There's no actual need for a resettable stream here and that's I didn't make it a part of the API.

Reply:
ah I see, thanks for clarifying.